### ISSUE TITLE
Crash of unknown cause

### ISSUE THREAD

**Issue by K4LCIFER** [xournalpp/xournalpp#4651]
**Affects versions :**
 - OS: Arch Linux
 - (Linux only) Desktop environment: Gnome 43.2 Wayland
 - Which version of libgtk do you use: 3.24.36
 - Version of Xournal++: 1.1.3 2dbd16d2
 - Installation method: Flatpak

**Describe the bug**
Crash of unknown cause

**To Reproduce**
n/a

**Expected behavior**
n/a

**Screenshots of Problem**
n/a

**Additional context**
xournalpp error logs:
```
Date: Sun Feb 12 22:53:45 2023
Error: signal 11
[bt]: (0) xournalpp(+0x200b85) [0x563857779b85]
[bt]: (1) /usr/lib/x86_64-linux-gnu/libc.so.6(+0x3f0c0) [0x7f5e7823f0c0]
[bt]: (2) xournalpp(_ZNK7Element7getTypeEv+0x4) [0x56385774aec4]
[bt]: (3) xournalpp(_ZNK12DocumentView11drawElementEP6_cairoP7Element+0x22) [0x563857770492]
[bt]: (4) xournalpp(_ZN12DocumentView9drawLayerEP6_cairoP5Layer+0x9b) [0x56385777064b]
[bt]: (5) xournalpp(_ZN12DocumentView8drawPageESt10shared_ptrI7XojPageEP6_cairobbbb+0xfe) [0x56385777109e]
[bt]: (6) xournalpp(_ZN10PreviewJob8drawPageEv+0x38a) [0x56385766d3fa]
[bt]: (7) xournalpp(_ZN10PreviewJob3runEv+0x34) [0x56385766d774]
[bt]: (8) xournalpp(_ZN9Scheduler17jobThreadCallbackEPS_+0xf3) [0x563857670343]
[bt]: (9) /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x87b59) [0x7f5e79ab7b59]
[bt]: (10) /usr/lib/x86_64-linux-gnu/libc.so.6(+0x8f1da) [0x7f5e7828f1da]
[bt]: (11) /usr/lib/x86_64-linux-gnu/libc.so.6(clone+0x44) [0x7f5e78317f44]


Try to get a better stracktrace...
[bt] #1 xournalpp(+0x2011a6) [0x56385777a1a6]
[bt] #2 /usr/lib/x86_64-linux-gnu/libc.so.6(+0x3f0c0) [0x7f5e7823f0c0]
[bt] #3 xournalpp(_ZNK7Element7getTypeEv+0x4) [0x56385774aec4]
[bt] #4 xournalpp(_ZNK12DocumentView11drawElementEP6_cairoP7Element+0x22) [0x563857770492]
[bt] #5 xournalpp(_ZN12DocumentView9drawLayerEP6_cairoP5Layer+0x9b) [0x56385777064b]
[bt] #6 xournalpp(_ZN12DocumentView8drawPageESt10shared_ptrI7XojPageEP6_cairobbbb+0xfe) [0x56385777109e]
[bt] #7 xournalpp(_ZN10PreviewJob8drawPageEv+0x38a) [0x56385766d3fa]
[bt] #8 xournalpp(_ZN10PreviewJob3runEv+0x34) [0x56385766d774]
[bt] #9 xournalpp(_ZN9Scheduler17jobThreadCallbackEPS_+0xf3) [0x563857670343]
[bt] #10 /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x87b59) [0x7f5e79ab7b59]
[bt] #11 /usr/lib/x86_64-linux-gnu/libc.so.6(+0x8f1da) [0x7f5e7828f1da]
[bt] #12 /usr/lib/x86_64-linux-gnu/libc.so.6(clone+0x44) [0x7f5e78317f44]
```

System Logs:
```
22:53:57 flatpak: F: Not sharing "/usr/share/icons" with sandbox: Path "/usr" is reserved by Flatpak
22:53:54 gnome-shell: Object .Gjs_ui_windowPreview_WindowPreview (0x55cfdb8f83c0), has been already disposed — impossible to get any property from it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
22:53:45 systemd: app-flatpak-com.github.xournalpp.xournalpp-127307.scope: Consumed 2min 25.966s CPU time.
22:53:45 xournalpp: Successfully saved document to "/home/<username>/.var/app/com.github.xournalpp.xournalpp/config/xournalpp/emergencysave.xopp"
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 sh: sh: line 1: addr2line: command not found
22:53:45 xournalpp: [Crash Handler] Wrote crash log to: /home/<username>/.var/app/com.github.xournalpp.xournalpp/cache/xournalpp/errorlogs/errorlog.20230212-225345.log
22:53:45 xournalpp: [Crash Handler] Wrote crash log to: /home/<username>/.var/app/com.github.xournalpp.xournalpp/cache/xournalpp/errorlogs/errorlog.20230212-225345.log
22:53:45 xournalpp: [Crash Handler] Crashed with signal 11
22:51:40 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:49:48 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:46:16 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:33:56 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:31:15 xournalpp: Error loading theme icon 'go-up' for stock: Icon 'go-up' not present in theme Adwaita
22:30:10 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:29:57 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:29:52 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:29:33 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:29:16 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:29:11 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
22:28:13 xournalpp: Error loading theme icon 'go-down' for stock: Icon 'go-down' not present in theme Adwaita
```

**Comment by K4LCIFER** [comment]
Possibly related to #4600?

**Comment by rolandlo** [comment]
> Possibly related to #4600?

Yes, it looks like the same issue. Maybe the [nightly build](https://github.com/xournalpp/xournalpp/releases/tag/nightly) 1.1.3+dev doesn't have it. Could you please try?

**Comment by K4LCIFER** [comment]
Is there a way for me to force the crash consistently? For me this is essentially a random event that happens quite infrequently. It would be very difficult to draw a conclusion from switching to 1.1.3+dev. This is perhaps a once or twice a month kind of thing.

**Comment by rolandlo** [comment]
We don't know how to force that crash. Maybe it's some data race.

**Comment by bhennion** [comment]
I'd say it is most likely a data race. There are plenty of place in the code where the document is modified without its mutex being locked. It's often a sneaky thing, with a class only remembering a pointer to a layer, and modifying this layer without locking the document.

The most frequent occurence of this is actually in StrokeHandler (the stroke is added to the document while the doc is not locked). This is fixed in #4158 (to be merged soon, hopefully).

**Comment by bhennion** [comment]
Closing in favor of the duplicate #4917
